tools/rsdl/dotnet/rsdl.models.transformation/ModelBuilder.cs

```diff
@@ -143,33 +146,41 @@ private void AddProperty(EdmStructuredType edmType, RdmProperty prop)
 
         private void AddFunction(RdmStructuredType rdmType, RdmFunction func)
         {
-            var isFunction = ! func.Annotations.Any(a => a is ActionAnnotation);
+            var isFunction = !func.Annotations.Any(a => a is ActionAnnotation);
             var edmTypeRef = func.ReturnType != null ? MakeTypeReference(func.ReturnType) : null;
             // TODO: check that a function has a return type
-            var operation =  isFunction ?
+            var edmOperation = isFunction ?
                 (EdmOperation)new EdmFunction(namespaceName, func.Name, edmTypeRef, true, null, true) :
-                (EdmOperation)new EdmAction(namespaceName, func.Name, edmTypeRef, true, null);            
-            edmModel.AddElement(operation);
+                (EdmOperation)new EdmAction(namespaceName, func.Name, edmTypeRef, true, null);
+            edmModel.AddElement(edmOperation);
 
             // TODO: add binding parameter
             var self = MakeTypeReference(new RdmTypeReference(rdmType.Name));
-            operation.AddParameter(new EdmOperationParameter(operation, "this", self));
+            edmOperation.AddParameter(new EdmOperationParameter(edmOperation, "this", self));
 
             foreach (var param in func.Parameters)
             {
                 var paramType = MakeTypeReference(param.PropType);
-                operation.AddParameter(new EdmOperationParameter(operation, param.Name, paramType));
+                var edmParameter = new EdmOperationParameter(edmOperation, param.Name, paramType);
+                edmOperation.AddParameter(edmParameter);
+                if (param.IsOptional)
+                {
+                    // https://github.com/oasis-tcs/odata-vocabularies/blob/master/vocabularies/Org.OData.Core.V1.md#OptionalParameter
+                    var annotation = new EdmVocabularyAnnotation(edmParameter, CoreVocabularyModel.OptionalParameterTerm, new EdmBooleanConstant(true));
+                    annotation.SetSerializationLocation(edmModel, EdmVocabularyAnnotationSerializationLocation.Inline);
+                    edmModel.AddVocabularyAnnotation(annotation);
+                }
             }
         }
 
         private void AddService(RdmService service)
         {
             var containerName = "default";
-            var container = (EdmEntityContainer) edmModel.EntityContainer ?? edmModel.AddEntityContainer(namespaceName, containerName);
+            var container = (EdmEntityContainer)edmModel.EntityContainer ?? edmModel.AddEntityContainer(namespaceName, containerName);
 
             foreach (var item in service.Items)
             {
-                switch(item)
+                switch (item)
```

Comment:
WDYT about https://github.com/dotnet/format. 
Adding linking on github actions will ensure format is good

Reply:
Interesting idea. This is different from the IDE (VS code) doing the formatting on save, right?
I am willing to try it but doesn't it mean that a pull request might still have wrongly formatted code and it just gets formatted after the PR is merged? 
Or is the idea that the github action actually commits the changed made by the tool?

Reply:
Ther might be a way to always run this tool at compile time which would make this independent from any editor. Will investigate.

Reply:
Generally it will save time if the build fails if you do not confirm to standards. 
It is great tool to streamline reviews. Like usual elements like tests, lint, format, and code style